### PR TITLE
Fix for snat-uuid missing in ep file

### PR DESCRIPTION
**Pull request by akhilamohanan**
Added indexMutex lock while updating snat-uuid in epfile

### CODE REVIEW

**Comment by akhilamohanan** [comment]
Testing is in progress

**Comment by akhilamohanan** [comment]
Tests done:
pytests 
stress test (by deleting host-agent in an interval of 5 min)

Built hostagent images with the patch:
quay.io/noirolabs/aci-containers-host:snat-uuid  (without enabling race flag)
quay.io/noirolabs/aci-containers-host:snat-uuid-race (by enabling race flag)

File: pkg/hostagent/snats.go [hunk 1]
```diff
@@ -1200,15 +1206,20 @@ func (agent *HostAgent) isPolicyNameSpaceMatches(policyName string, namespace st
...
-	agent.indexMutex.Unlock()
 	if check {
-		return val.PlcyUuids
-
-	} else {
-		return []string{}
+		err := util.DeepCopyObj(val, localInfo)
+		if err != nil {
+			agent.log.Error(err.Error())
+			return plcyUuids
```
Comment (tanyatukade) [inline pkg/hostagent/snats.go:1218]: Why are we returning the plcyUuids even if the deep copy fails? We should be returning nil. In general it is best practice to have the return value([]string) and the error in the func definition, so we can return both. For this case, when there is an err we return nil, err and when the func runs sucessfully we return plcyUuids, nil

Comment (rupeshsahuoc) [reply pkg/hostagent/snats.go]: addressed

**Pull request closed without merging**